### PR TITLE
[FLOC-1510] Centos7 tutorial environment

### PR DESCRIPTION
**Pull request by adamtheturtle**
Fixes https://clusterhq.atlassian.net/browse/FLOC-1510

I *think* that all the test failures are unrelated. I also think that the test builder's names like acceptance/fedora are a separate issue. This has a mixture of Python and shell, but I chose this so that bootstrap.sh could be similar to the dev box's bootstrap code.

### CODE REVIEW

File: vagrant/dev/bootstrap.sh [hunk 1]
```diff
@@ -20,6 +20,7 @@ yum install -y \
 	docker \
 	python-devel python-tox \
 	python-virtualenv python-virtualenvwrapper python-pip \
-	libffi-devel \
+  enchant \
```
Comment (adamtheturtle) [inline vagrant/dev/bootstrap.sh]: To do this branch I needed to `pip install -e .[doc,dev]` (or at least, I did that) on the dev box. These dependencies were necessary.

File: vagrant/dev/bootstrap.sh [hunk 2]
```diff
@@ -20,6 +20,7 @@ yum install -y \
 	docker \
 	python-devel python-tox \
 	python-virtualenv python-virtualenvwrapper python-pip \
-	libffi-devel \
+	enchant \
```
Comment (adamtheturtle) [inline vagrant/dev/bootstrap.sh]: To do this branch I needed to `pip install -e .[doc,dev]` (or at least, I did that) on the dev box. These dependencies were necessary.

File: vagrant/tutorial/post-reboot-bootstrap.py [hunk 3]
```diff
@@ -18,16 +18,14 @@
...
 
+# This uses development repo, because the marketing one has not been set up yet
+# this should be changed before merging.
 clusterhq_repo_url = (
-    'https://storage.googleapis.com/'
-    'archive.clusterhq.com/'
-    'fedora/clusterhq-release%s.noarch.rpm') % (rpm_dist,)
+    'https://s3.amazonaws.com/'
+    'clusterhq-archive/'
+    'development/centos/'
```
Comment (adamtheturtle) [inline vagrant/tutorial/post-reboot-bootstrap.py]: This is just temporary. The repo will be in /centos, not /development/centos

**Comment by adamtheturtle** [comment]
Because of https://github.com/ClusterHQ/flocker/pull/1210/files#r26212040 this cannot be merged until work has been done in another branch. However I think that it is ready for review and that can be changed later.

File: docs/gettinginvolved/infrastructure/vagrant.rst [hunk 4]
```diff
@@ -12,11 +12,11 @@ Boxes
 There are several vagrant boxes.
 
 Development Box (:file:`vagrant/dev`)
-   The box is initialized with the yum repositories for ZFS and for dependencies not available in Fedora and installs all the dependencies.
+   The CentOS box is initialized with the yum repositories for ZFS.
```
Comment (wallrj) [inline docs/gettinginvolved/infrastructure/vagrant.rst]: s/The/This/

File: docs/gettinginvolved/infrastructure/vagrant.rst [hunk 5]
```diff
@@ -25,7 +25,7 @@ Building
 
 To build one of the above boxes, install the necessary Vagrant plugins and run the :file:`build` script in the corresponding directory.
 
-To build the development box, install the necessary Vagrant plugins as follows:
+To build the development or tutorial box, install the necessary Vagrant plugins as follows:
```
Comment (wallrj) [inline docs/gettinginvolved/infrastructure/vagrant.rst]: This line (or its position) doesn't make sense to me. 

Surely you have to install the required plugins before building the `box`?

File: vagrant/tutorial/Vagrantfile [hunk 6]
```diff
@@ -4,10 +4,42 @@
...
+end
+
+=begin
+  The following error may occur when running `vagrant up`, potentially because
+  of a problem with some VirtualBox versions.
+
+    # Failed to mount folders in Linux guest. This is usually because
+    # the "vboxsf" file system is not available. Please verify that
+    # the guest additions are properly installed in the guest and
+    # can work properly. The comm du yand attempted was:
```
Comment (wallrj) [inline vagrant/tutorial/Vagrantfile]: "comm du yand" ?

File: vagrant/tutorial/Vagrantfile [hunk 7]
```diff
@@ -4,10 +4,42 @@
...
+    #
+    # mount -t vboxsf -o uid=`id -u vagrant`,gid=`getent group vagrant | cut -d: -f3` vagrant /vagrant
+    # mount -t vboxsf -o uid=`id -u vagrant`,gid=`id -g vagrant` vagrant /vagrant
+    #
+    # The error output from the last command was:
+    #
+    # /sbin/mount.vboxsf: mounting failed with the error: No such device
```
Comment (wallrj) [inline vagrant/tutorial/Vagrantfile]: This seems a rather long comment to leave in the final Vagrantfile.
And I'm still not sure of the circumstances where this problem might occur.
Is it a MacOSX thing? If so, just say that and link to a GitHub issue.

**Comment by wallrj** [comment]
I noticed this warning / error when I attempted to build the box.

```
==> default: Machine booted and ready!
^[[6~GuestAdditions seems to be installed (4.3.24) correctly, but not running.
Starting vboxadd (via systemctl):  Job for vboxadd.service failed. See 'systemctl status vboxadd.service' and 'journalctl -xn' for details.
[FAILED]
Removing existing VirtualBox non-DKMS kernel modules[  OK  ]
Building the VirtualBox Guest Additions kernel modules
The gcc utility was not found. If the following module compilation fails then
this could be the reason and you should try installing it.

The headers for the current running kernel were not found. If the following
module compilation fails then this could be the reason.
The missing package can be probably installed with
yum install kernel-devel-3.10.0-123.20.1.el7.x86_64

Building the main Guest Additions module[FAILED]
(Look at /var/log/vboxadd-install.log to find out what went wrong)
Doing non-kernel setup of the Guest Additions[  OK  ]
Restarting VM to apply changes...

```

File: vagrant/tutorial/bootstrap.sh [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+
+# This script performs the steps to build the base flocker-tutorial box until
+# the box must be rebooted.
+
+set -e
+
+yum update -y
+# Install useful yum repos
+yum install -y https://s3.amazonaws.com/archive.zfsonlinux.org/epel/zfs-release$(rpm -E %dist).noarch.rpm
+yum install -y epel-release
```
Comment (wallrj) [inline vagrant/tutorial/bootstrap.sh]: I wonder how this ties in with @tomprince 's Centos7 installation docs / acceptance testing changes in #1207 :
 * https://github.com/ClusterHQ/flocker/pull/1207/files#diff-bd81b442286984f45ec7fe47f8aa0584R22

The `run-acceptance-tests` tool has a `--provisioner=vagrant`  option, which (I think) installs these repos too, so perhaps those tasks can be re-used here. Maybe create a followup issue.

**Comment by wallrj** [comment]
And the build script ultimately failed, despite having the necessary plugins installed:
 
```
==> default: Mounting shared folders...
    default: /vagrant => /home/richard/projects/HybridLogic/flocker/vagrant/tutorial

Failed to mount folders in Linux guest. This is usually because
the "vboxsf" file system is not available. Please verify that
the guest additions are properly installed in the guest and
can work properly. The command attempted was:

mount -t vboxsf -o uid=`id -u vagrant`,gid=`getent group vagrant | cut -d: -f3` vagrant /vagrant
mount -t vboxsf -o uid=`id -u vagrant`,gid=`id -g vagrant` vagrant /vagrant

The error output from the last command was:

/sbin/mount.vboxsf: mounting failed with the error: No such device

Failed 1: vagrant up
```

I think this needs some further work.

**Comment by wallrj** [comment]
```
(1510)(vagrant-tutorial-FLOC-1510 ? =)[~/.../vagrant/tutorial]$ VBoxHeadless  --version
Oracle VM VirtualBox Headless Interface 4.3.24
(C) 2008-2015 Oracle Corporation
All rights reserved.

4.3.24r98716

(1510)(vagrant-tutorial-FLOC-1510 ? =)[~/.../vagrant/tutorial]$ vagrant --version
Vagrant 1.7.2
```

File: vagrant/tutorial/Vagrantfile (hunk 7 above)
Comment (adamtheturtle) [inline vagrant/tutorial/Vagrantfile]: > This seems a rather long comment to leave in the final Vagrantfile.

This isn't really the final Vagrantfile. There is another Vagrantfile which uses the box build from this file.

> And I'm still not sure of the circumstances where this problem might occur.

Neither am I - it seems to happen with some versions of VirtualBox and Vagrant not mixing well, but I can't be sure and thought that investigation was not worth it with a not-so-bad mitigation plugin here.

> Is it a MacOSX thing? If so, just say that and link to a GitHub issue.

Unsure as above.

File: vagrant/tutorial/bootstrap.sh (hunk 8 above)
Comment (adamtheturtle) [inline vagrant/tutorial/bootstrap.sh]: That *looks* like it uses the development kernel while this uses a released kernel.

**Comment by adamtheturtle** [comment]
I think that the issue you experienced is fixed.

Trying to build the box locally fails though, with error similar to:
```
==> default: subprocess.CalledProcessError: Command '['yum', 'install', '-y', 'clusterhq-flocker-node-0.3.2+doc1-1.2052.g56a4678']' returned non-zero exit status 1
```

This is because that package is not available at http://build.clusterhq.com/results/omnibus/vagrant-tutorial-FLOC-1510/centos-7/. However, I think that this is because the packages must be built before the Vagrant boxes are built, as Buildbot does. Note that Buildbot successfully built this box. I suggest that the instruction to run `./build` is removed and replaced with instructions to get this box from Buildbot.

**Comment by adamtheturtle** [comment]
I have to look into the issue with running the acceptance tests:

```
Traceback (most recent call last):
  File "/home/buildslave/fedora-vagrant/flocker-acceptance-vagrant-fedora-20/build/admin/run-acceptance-tests", line 13, in <module>
    main(sys.argv[1:], top_level=TOPLEVEL, base_path=BASEPATH)
  File "/home/buildslave/fedora-vagrant/flocker-acceptance-vagrant-fedora-20/build/admin/acceptance.py", line 367, in main
    nodes = runner.start_nodes()
  File "/home/buildslave/fedora-vagrant/flocker-acceptance-vagrant-fedora-20/build/admin/acceptance.py", line 149, in start_nodes
    commands=task_pull_docker_images()
  File "/home/buildslave/fedora-vagrant/flocker-acceptance-vagrant-fedora-20/build/flocker/provision/_install.py", line 100, in run_with_fabric
    handlers[type(command)](command)
  File "/home/buildslave/fedora-vagrant/flocker-acceptance-vagrant-fedora-20/build/flocker/provision/_install.py", line 86, in <lambda>
    Run: lambda e: run(e.command),
  File "/home/buildslave/fedora-vagrant/flocker-acceptance-vagrant-fedora-20/venv/lib/python2.7/site-packages/fabric/network.py", line 647, in host_prompting_wrapper
    return func(*args, **kwargs)
  File "/home/buildslave/fedora-vagrant/flocker-acceptance-vagrant-fedora-20/venv/lib/python2.7/site-packages/fabric/operations.py", line 1048, in run
    shell_escape=shell_escape)
  File "/home/buildslave/fedora-vagrant/flocker-acceptance-vagrant-fedora-20/venv/lib/python2.7/site-packages/fabric/operations.py", line 915, in _run_command
    channel=default_channel(), command=wrapped_command, pty=pty,
  File "/home/buildslave/fedora-vagrant/flocker-acceptance-vagrant-fedora-20/venv/lib/python2.7/site-packages/fabric/state.py", line 397, in default_channel
    chan = _open_session()
  File "/home/buildslave/fedora-vagrant/flocker-acceptance-vagrant-fedora-20/venv/lib/python2.7/site-packages/fabric/state.py", line 389, in _open_session
    return connections[env.host_string].get_transport().open_session()
  File "/home/buildslave/fedora-vagrant/flocker-acceptance-vagrant-fedora-20/venv/lib/python2.7/site-packages/fabric/network.py", line 159, in __getitem__
    self.connect(key)
  File "/home/buildslave/fedora-vagrant/flocker-acceptance-vagrant-fedora-20/venv/lib/python2.7/site-packages/fabric/network.py", line 151, in connect
    user, host, port, cache=self, seek_gateway=seek_gateway)
  File "/home/buildslave/fedora-vagrant/flocker-acceptance-vagrant-fedora-20/venv/lib/python2.7/site-packages/fabric/network.py", line 465, in connect
    raise NetworkError("Host key for %s did not match pre-existing key! Server's key was changed recently, or possible man-in-the-middle attack." % host, e)
fabric.exceptions.NetworkError: Host key for 172.16.255.240 did not match pre-existing key! Server's key was changed recently, or possible man-in-the-middle attack.
program finished with exit code 1
```

**Comment by tomprince** [comment]
> I have to look into the issue with running the acceptance tests:

This is https://clusterhq.atlassian.net/browse/FLOC-1527.

**Comment by adamtheturtle** [comment]
@tomprince Thanks, I'll make that issue a blocker to this one then. But I guess that this can still be reviewed so I'll leave it open.

File: vagrant/tutorial/post-reboot-bootstrap.py [hunk 9]
```diff
@@ -18,16 +18,14 @@
...
 # Make it possible to install flocker-node
 rpm_dist = check_output(['rpm', '-E', '%dist']).strip()
-zfs_repo_url = (
-    'https://s3.amazonaws.com/'
-    'archive.zfsonlinux.org/'
-    'fedora/zfs-release%s.noarch.rpm') % (rpm_dist,)
-check_call(['yum', 'install', '-y',  zfs_repo_url])
 
+# This uses development repo, because the marketing one has not been set up yet
+# this should be changed before merging.
```
Comment (exarkun) [inline vagrant/tutorial/post-reboot-bootstrap.py]: Does it still?  It doesn't **look** like a development repository.

Comment (adamtheturtle) [inline vagrant/tutorial/post-reboot-bootstrap.py]: No, I changed it but did not remove the note.

**Comment by exarkun** [comment]
Trying this out, it seems I get the same error as @wallrj 

```
    default: /vagrant => /home/jean-paul/Projects/flocker/review/vagrant/tutorial
Failed to mount folders in Linux guest. This is usually because
the "vboxsf" file system is not available. Please verify that
the guest additions are properly installed in the guest and
can work properly. The command attempted was:

mount -t vboxsf -o uid=`id -u vagrant`,gid=`getent group vagrant | cut -d: -f3` vagrant /vagrant
mount -t vboxsf -o uid=`id -u vagrant`,gid=`id -g vagrant` vagrant /vagrant

The error output from the last command was:

/sbin/mount.vboxsf: mounting failed with the error: No such device

Failed 1: vagrant up
```

**Comment by adamtheturtle** [comment]
@exarkun :cry: That happened locally for me but stopped happening with the later commits.

However, does this matter? I proposed above:

> I suggest that the instruction to run ./build is removed and replaced with instructions to get this box from Buildbot.

**Comment by exarkun** [comment]
Reading earlier parts of the build log, it looks like building guest additions fails.  Reading the log on the VM:

```
/tmp/vbox.0/r0drv/linux/memobj-r0drv-linux.c: In function ‘rtR0MemObjNativeMapUser’:
/tmp/vbox.0/r0drv/linux/memobj-r0drv-linux.c:1542:26: error: ‘struct mm_struct’ has no member named ‘numa_next_reset’
                 pTask->mm->numa_next_reset = jiffies + 0x7fffffffffffffffUL;
                          ^
make[2]: *** [/tmp/vbox.0/r0drv/linux/memobj-r0drv-linux.o] Error 1
```

I don't know if this is related to the overall box build failing, though.

**Comment by exarkun** [comment]
> However, does this matter? I proposed above:
> > I suggest that the instruction to run ./build is removed and replaced with instructions to get this box from Buildbot.

Maybe not.  However, all things equal, it would be better if it were possible to build the box locally instead of having to rely on buildbot to build it.

Do you have any idea why it works on buildbot and not elsewhere?  If not, it's probably at least worth understanding that difference.  Otherwise when it eventually breaks on buildbot we can't build the box at all.

**Comment by adamtheturtle** [comment]
As I mentioned earlier, it maybe wouldn't even be possible even if this problem was resolved because building the box relies on packages being available from Buildbot.

I don't know why this works on Buildbot and my Mac and not your machine or Richard's (although Richard hasn't tested this since the "fix" was applied).

```
(flocker)~/Documents/ClusterHQ/flocker/vagrant/tutorial$ vagrant version
Installed Version: 1.7.2
Latest Version: 1.7.2

You're running an up-to-date version of Vagrant!
(flocker)~/Documents/ClusterHQ/flocker/vagrant/tutorial$ VBoxHeadless  --version
Oracle VM VirtualBox Headless Interface 4.3.24
(C) 2008-2015 Oracle Corporation
All rights reserved.

4.3.24r98716
```

**Comment by exarkun** [comment]
> As I mentioned earlier, it maybe wouldn't even be possible even if this problem was resolved because building the box relies on packages being available from Buildbot.

Therefore we should fix this so it's possible to build the box locally.  Right?

**Comment by adamtheturtle** [comment]
> Therefore we should fix this so it's possible to build the box locally. Right?

Well, this is an existing issue, I get:

```
==> default: Error: Nothing to do
==> default: Traceback (most recent call last):
==> default:   File "/tmp/vagrant-shell", line 63, in <module>
==> default:
==> default: check_call(['yum', 'install', '-y'] + branch_opt + [package])
==> default:   File "/usr/lib64/python2.7/subprocess.py", line 542, in check_call
==> default:
==> default: raise CalledProcessError(retcode, cmd)
==> default: subprocess
==> default: .
==> default: CalledProcessError
==> default: : Command '['yum', 'install', '-y', '--enablerepo=clusterhq-build', 'clusterhq-flocker-node-0.3.2+doc1-1.2197.g64e9f8e']' returned non-zero exit status 1
``` 

running `./build --branch vagrant-tutorial-FLOC-1510` on `master`. I'm not convinced that there is a big advantage in doing this locally. I'm not saying that there is none, but I think that this can at least be punted to a further issue. If you agree, I can create that issue (something like "Make it possible to build the tutorial box locally").

**Comment by exarkun** [comment]
> Well, this is an existing issue, I get:

That seems to be a different issue.  It's a `yum install` command failing because the indicated package doesn't exist.  The error @wallrj and I encountered is an error that seems to have to do with mounting the shared `/vagrant` folder?  Or did I miss something?

**Comment by exarkun** [comment]
I upgraded from VirtualBox 4.3.10 to 4.3.24 and by specifying a known-good version to the build script managed to get a successful build.

So I think maybe the plugin that's supposed to handle host/guest version mismatch doesn't actually work reliably and we should just specify an exact version of VirtualBox to use when doing these builds.

File: vagrant/tutorial/Vagrantfile (hunk 7 above)
Comment (adamtheturtle) [inline vagrant/tutorial/Vagrantfile]: I've added the line "Known working versions include Vagrant 1.7.2 and VirtualBox 4.3.24r98716."

**Comment by adamtheturtle** [comment]
I have tested that this works for me with 4.3.26 too. I have added a section saying that Buildbot builds the boxes and noted the known working set of versions. Therefore I think that this is ready for another review. As mentioned above https://clusterhq.atlassian.net/browse/FLOC-1527 should be a blocker to merging this but not reviewing it.

File: vagrant/dev/bootstrap.sh (hunk 2 above)
Comment (adamtheturtle) [inline vagrant/dev/bootstrap.sh]: Because this is unrelated and came up in the release review I am separating this out - see FLOC-1565

**Comment by itamarst** [comment]
Tests are failing in ways that might be significant.

File: vagrant/tutorial/bootstrap.sh [hunk 10]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+# This script performs the steps to build the base flocker-tutorial box until
+# the box must be rebooted.
+
+set -e
+
+yum update -y
+# Install useful yum repos
+yum install -y https://s3.amazonaws.com/archive.zfsonlinux.org/epel/zfs-release$(rpm -E %dist).noarch.rpm
```
Comment (itamarst) [inline vagrant/tutorial/bootstrap.sh]: Aren't we using our own custom ZFS packages for now?

**Comment by itamarst** [comment]
Resubmit once buildbot is green.

**Comment by adamtheturtle** [comment]
The tutorial box builder needs to be triggered by the CentOS package builder, not the Fedora one. The `flocker/acceptance/vagrant/fedora-20` builder name should be changed to references `centos-7` and the directory in `flocker/admin/vagrant-acceptance-targets` should be renamed appropriately.

File: vagrant/tutorial/Vagrantfile [hunk 11]
```diff
@@ -4,19 +4,63 @@
...
+  config.vm.provision :shell, :path => "post-reboot-bootstrap.py", :privileged => true,
       :args => [
           ENV.fetch('FLOCKER_RPM_VERSION', ''),
           ENV.fetch('FLOCKER_BRANCH', ''),
           ENV.fetch('FLOCKER_BUILD_SERVER', '')
       ]
-  # Prepare caches
-  config.vm.provision :shell, :path => "cache.sh", :privileged => true
+
+  config.vm.provision "docker", images: [
```
Comment (adamtheturtle) [inline vagrant/tutorial/Vagrantfile]: Not a new issue but it seems to me that for the simplicity of having one box we are causing tutorial users to download a box with `redis` and `flask` and getting started users to download a box with `mongodb`. If you think that there is value in having two boxes, create a follow up.

File: vagrant/tutorial/Vagrantfile [hunk 12]
```diff
@@ -4,19 +4,63 @@
...
+
   # Cleanup
   config.vm.provision :shell, :path => "../cleanup.sh", :privileged => true
+  #
+  # Don't use a shared folder.
+  # - It isn't used during the build or after.
+  # - The vguest plugin tries to compile a new vboxsf module, but
+  #   fails on the first boot, since it can't install the corresponding
+  #   kernel-devel headers.
+  config.vm.synced_folder ".", "/vagrant", disabled: true
```
Comment (adamtheturtle) [inline vagrant/tutorial/Vagrantfile]: If there's now no shared folder do we care about the guest additions?

File: vagrant/tutorial/post-reboot-bootstrap.py [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+for service in ['flocker-control-api', 'flocker-control-agent']:
+    check_call(['firewall-cmd', '--permanent', '--add-service', service])
+
+# Make it easy to authenticate as root
+check_call(['mkdir', '-p', '/root/.ssh'])
+check_call(
+    ['cp', os.path.expanduser('~vagrant/.ssh/authorized_keys'), '/root/.ssh'])
+
+# Configure GRUB2 to boot kernel with elevator=noop to workaround
+# https://clusterhq.atlassian.net/browse/FLOC-235
```
Comment (adamtheturtle) [inline vagrant/tutorial/post-reboot-bootstrap.py]: That issue is marked as DONE which makes me think that this might not be necessary now - and the comments talk about a Fedora fix.

File: vagrant/tutorial/post-reboot-bootstrap.py [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+check_call(['systemctl', 'unmask', 'firewalld'])
+check_call(['systemctl', 'enable', 'firewalld'])
+check_call(['systemctl', 'start', 'firewalld'])
+# We need to open the firewall for the flocker-control
+for service in ['flocker-control-api', 'flocker-control-agent']:
+    check_call(['firewall-cmd', '--permanent', '--add-service', service])
+
+# Make it easy to authenticate as root
+check_call(['mkdir', '-p', '/root/.ssh'])
+check_call(
```
Comment (adamtheturtle) [inline vagrant/tutorial/post-reboot-bootstrap.py]: Can this be done without `check_call`?

File: vagrant/tutorial/post-reboot-bootstrap.py [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+# We need to unmask it, since the base box has it masked.
+check_call(['systemctl', 'unmask', 'firewalld'])
+check_call(['systemctl', 'enable', 'firewalld'])
+check_call(['systemctl', 'start', 'firewalld'])
+# We need to open the firewall for the flocker-control
+for service in ['flocker-control-api', 'flocker-control-agent']:
+    check_call(['firewall-cmd', '--permanent', '--add-service', service])
+
+# Make it easy to authenticate as root
+check_call(['mkdir', '-p', '/root/.ssh'])
```
Comment (adamtheturtle) [inline vagrant/tutorial/post-reboot-bootstrap.py]: Can this be done without check_call?

File: vagrant/tutorial/post-reboot-bootstrap.py [hunk 16]
```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/python
+
+# This script builds the base flocker-dev box.
```
Comment (adamtheturtle) [inline vagrant/tutorial/post-reboot-bootstrap.py]: tutorial box

File: vagrant/tutorial/post-reboot-bootstrap.py (hunk 15 above)
Comment (adamtheturtle) [inline vagrant/tutorial/post-reboot-bootstrap.py]: For the other mkdir calls too

File: vagrant/tutorial/post-reboot-bootstrap.py [hunk 17]
```diff
@@ -0,0 +1,112 @@
...
+    # The buildserver doesn't build dirty versions,
+    # so strip that.
+    if rpm_version.endswith('.dirty'):
+        rpm_version = rpm_version[:-len('.dirty')]
+    package = 'clusterhq-flocker-node-%s' % (rpm_version,)
+else:
+    package = 'clusterhq-flocker-node'
+
+# Install flocker-node
+check_call(['yum', 'install', '-y'] + branch_opt + [package])
```
Comment (adamtheturtle) [inline vagrant/tutorial/post-reboot-bootstrap.py]: :question: Use `def yum_install(*packages):` as in the other script.

**Comment by adamtheturtle** [comment]
The main concerns are that the manual tutorial works, and that the ZFS acceptance tests pass on Buildbot. As the acceptance tests are passing and I think you did the manual testing (?) I think that this is good to go and I'm pretty happy saying that because the work that I did was reviewed by others already. Resubmit if you want someone else to look at this.
